react/features/base/conference/middleware.js

```diff
@@ -93,27 +95,46 @@ function _connectionEstablished(store, next, action) {
  */
 function _pinParticipant(store, next, action) {
     const state = store.getState();
+    const { conference } = state['features/base/conference'];
     const participants = state['features/base/participants'];
     const id = action.participant.id;
     const participantById = getParticipantById(participants, id);
     let pin;
 
-    // The following condition prevents signaling to pin local participant. The
-    // logic is:
+    const shouldEmitToLegacyApp = typeof APP !== 'undefined';
+
+    if (shouldEmitToLegacyApp) {
+        const pinnedParticipant = getPinnedParticipant(participants);
+        const actionName = action.participant.id ? 'pinned' : 'unpinned';
+        let videoType;
+
+        if ((participantById && participantById.local)
+            || (!id && pinnedParticipant && pinnedParticipant.local)) {
+            videoType = 'local';
+        } else {
+            videoType = 'remote';
+        }
+
+        JitsiMeetJS.analytics.sendEvent(
+                `${actionName}.${videoType}`,
+                { value: conference.getParticipantCount() });
+    }
+
+    // The following condition prevents signaling to pin local participant and
+    // shared videos. The logic is:
     // - If we have an ID, we check if the participant identified by that ID is
-    //   local.
+    //   local or a shared video.
     // - If we don't have an ID (i.e. no participant identified by an ID), we
     //   check for local participant. If she's currently pinned, then this
     //   action will unpin her and that's why we won't signal here too.
     if (participantById) {
-        pin = !participantById.local;
+        pin = !participantById.local && !participantById.isSharedVideo;
```

Comment:
So now this check for 'isSharedVideo' is instead of checking for video container type ?
I wonder if we can come up with something more generic, because I imagine we may have other video thumbnails than the shared video (we used to have "Prezi" thumbnail some time ago). I see that also in 'isLonelyCall' method this field is used to filter out real participants, so maybe we could have a boolean or a method which tells exactly that: "isRealParticipant" ? It's value could be evaluate based on 'isSharedVideo' for starters.

We also have Jibri participants which are hidden in the UI. I don't know if those are filtered out earlier, before triggering participant joined, so that they never make it to the list of participants('features/base/participants').